### PR TITLE
Remove hardcoded hotkey references from production, support powers, and viewport.

### PR DESCRIPTION
**Pull request by pchote**
Part 3 of my campaign to unhardcode and move gameplay-specific hotkeys to mod code.
This follows on from #13706 and #13711.

This converts all the remaining hotkey consumers except for `WorldCommandWidget` and the hardcoded "dev" hotkeys.  Those two require more drastic changes that will come in their own PRs.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs]: Lines hotkeyText, GetText, Bounds can be inside IsValid() block.

Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs]: var nameWidth = font.Measure(name).X; could be used and reused below like in ProductionTooltipLogic.

File: OpenRA.Mods.Common/Widgets/ProductionTabsWidget.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/ProductionTabsWidget.cs]: It could be Previous, Next like below and in yaml.

File: OpenRA.Mods.Common/Lint/CheckChromeHotkeys.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Lint/CheckChromeHotkeys.cs]: This works fine with HotkeyCount=0, but with missing HotkeyPrefix it passes lint but produces strange/wrong hotkeys. Ideally HotkeyPrefix should be mandatory for count>0.

**Review by rob-v — COMMENTED** [review]
mostly optional nits.

**Comment by pchote** [comment]
Updated as requested, and added a few more cleanups on top of your suggestions to the support/production tooltip logic.

File: OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs]: SupportPowers/SupportPalette instead of ProductionPalette here.

File: OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs]: You used `prefix = ""` in SupportPowersWidget, ProductionPaletteWidget.

File: OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs]: BookmarkRestoreKeyPrefix here.

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs]: nameFont.Measure(hotkeyText).X should be kept here?

**Review by rob-v — COMMENTED** [review]
small issues

**Comment by pchote** [comment]
D'oh.  Fixed.

**Review by rob-v — APPROVED** [review]
lgtm 👍 

throw in LinterHotkeyNames is less convinient - emitError could be passed to use instead or added try/catch around LinterHotkeyNames call - to display only mesage and continue checking instead of stacktrace and 'crash'/stop on first occurrence.

**Comment by pchote** [comment]
Good idea, will do.

**Comment by pchote** [comment]
Done.  This also let me print the file and line numbers:
```
Testing Tiberian Sun mod MiniYAML...
Testing mod: Tiberian Sun
OpenRA.Utility(1,1): Error: ts|chrome/ingame-player.yaml:474 must define HotkeyPrefix if HotkeyCount > 0.
OpenRA.Utility(1,1): Error: common|chrome/editor.yaml:214 must define BookmarkRestoreKeyPrefix if BookmarkKeyCount > 0.
```

**Comment by rob-v** [comment]
Could you clarify the logic behind hotkeys defined in Widget vs Logic due to different configuration/implementation?

**Comment by pchote** [comment]
Keys that control or interact with a widget go on that widget.  Everything else should be implemented as stand-alone logic classes.

**Review by reaperrr — DISMISSED** [review]

**Comment by reaperrr** [comment]
Note: it doesn't seem to be caused by this PR (hence the +1), but next/prev tab (page u/down) don't work on my system, neither with this PR nor on bleed.

Everything else works, but I'll wait for a reply (and rebase, obviously) before I merge this.

**Comment by rob-v** [comment]
Page up/down works e.g. in TD to switch between all barracks,... iirc.

**Comment by reaperrr** [comment]
> Page up/down works e.g. in TD to switch between all barracks,... iirc.

Ah, I see. That works, so just needs a rebase then.

**Comment by pchote** [comment]
Rebased.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3b138dbf9bd34d319974cf8c92790a73878d6d8f...00ab906438192ab9831ea68647580601a1283314

**Pull request merged**
